src/BugBundle/Entity/Issue.php

```diff
@@ -0,0 +1,616 @@
+     * @ORM\OneToMany(targetEntity="BugBundle\Entity\IssueComment", mappedBy="issue")
+     */
+    private $comments;
+
+    /**
+     * @var integer
+     *
+     * @ORM\Column(name="type", type="integer")
+     */
+    private $type;
+
+    /**
+     * @var IssuePriority
+     * @ORM\ManyToOne(targetEntity="BugBundle\Entity\IssuePriority")
+     * @ORM\JoinColumn(name="priority_id", referencedColumnName="id", nullable=false)
+     */
+    private $priority;
+
+    /**
+     * @var IssueStatus
+     * @ORM\ManyToOne(targetEntity="BugBundle\Entity\IssueStatus")
+     * @ORM\JoinColumn(name="status_id", referencedColumnName="id", nullable=false)
+     */
+    private $status;
+
+    /**
+     * @var IssueResolution
+     * @ORM\ManyToOne(targetEntity="BugBundle\Entity\IssueResolution")
+     * @ORM\JoinColumn(name="resolution_id", referencedColumnName="id", nullable=false)
+     */
+    private $resolution;
+
+    /**
+     * @var User
+     * @ORM\ManyToOne(targetEntity="BugBundle\Entity\User")
+     * @ORM\JoinColumn(name="reporter_id", referencedColumnName="id", nullable=false)
+     */
+    private $reporter;
+
+    /**
+     * @var User
+     * @ORM\ManyToOne(targetEntity="BugBundle\Entity\User")
+     * @ORM\JoinColumn(name="assignee_id", referencedColumnName="id", nullable=false)
+     */
+    private $assignee;
+
+    /**
+     * @var Collection | User[]
+     * @ORM\ManyToMany(targetEntity="BugBundle\Entity\User", inversedBy="issues")
+     * @ORM\JoinTable(name="collaborators")
+     **/
+    private $collaborators;
+
+    /**
+     * @var Issue
+     * @ORM\ManyToOne(targetEntity="BugBundle\Entity\Issue", inversedBy="childrenIssues")
+     * @ORM\JoinColumn(name="parent_id", referencedColumnName="id", nullable=true, onDelete="CASCADE")
+     */
+    private $parentIssue;
+
+
+    /**
+     * @var Collection | Issue[]
+     * @ORM\OneToMany(targetEntity="BugBundle\Entity\Issue", mappedBy="parentIssue")
+     */
+    private $childrenIssues;
+
+    /**
+     * @var User
+     * @ORM\ManyToOne(targetEntity="BugBundle\Entity\Project", inversedBy="issues")
+     * @ORM\JoinColumn(name="project_id", referencedColumnName="id", nullable=false, onDelete="CASCADE")
+     */
+    private $project;
+
+    /**
+     * @var \DateTime
+     *
+     * @ORM\Column(name="created", type="datetime")
+     */
+    private $created;
+
+    /**
+     * @var \DateTime
+     *
+     * @ORM\Column(name="updated", type="datetime")
+     */
+    private $updated;
+
+    public function __construct()
+    {
+        $this->childrenIssues = new ArrayCollection();
+        $this->collaborators = new ArrayCollection();
+        $this->comments = new ArrayCollection();
+    }
+
+    public function __toString()
+    {
+        return $this->getIssueFullName() ? $this->getIssueFullName() : '';
+    }
+
+    private function getIssueFullName()
+    {
+
+        return ($this->code && $this->id && $this->summary) ? $this->code.'-'.$this->id.' '.$this->summary : '';
```

Comment:
only code is emtpy? only id or summary is empty? - all this cases are missing in tests...